src/AlevinUtils.cpp

```diff
@@ -378,21 +378,35 @@ namespace alevin {
         (bc.assign(read, pt.umiLength, pt.barcodeLength), true) : false;
     }
     template <>
-    bool extractBarcode<apt::InDrop>(std::string& read, 
+    bool extractBarcode<apt::InDropV2>(std::string& read, 
                                      std::string& read2, 
-                                     apt::InDrop& pt, std::string& bc){
-      (void)read2;
-      std::string::size_type index = read.find(pt.w1);
-      if (index == std::string::npos){
-        return false;
+                                     apt::InDropV2& pt, std::string& bc){
+      (void)read;
+      if(read2.length() >= (pt.w1Length + pt.barcodeLength + pt.umiLength)) {
+      pt.w1Pos = read2.find(pt.w1);
+      if (pt.w1Pos == std::string::npos){
+        bool found = false;
+        for( int i = 8; i <= 11; i++){
+          if (hammingDistance(pt.w1, read2.substr(i,pt.w1Length)) <= pt.maxHammingDist) {
+            pt.w1Pos = i;
+            found = true;
+            break;
+          }
+        }
+        if (!found) {return false;}
       }
-      bc = read.substr(0, index);
-      if(bc.size()<8 or bc.size()>12){
+      bc = read2.substr(0, pt.w1Pos);
+      if(bc.size()<8 or bc.size()>11){
         return false;
       }
       uint32_t offset = bc.size()+pt.w1.size();
-      bc += read.substr(offset, offset+8);
+      bc += read2.substr(offset, 8);
+      bc += std::string(pt.barcodeLength - bc.size(), 'A');
```

Comment:
@Gaura : Do we want to consider "unique padding" just as we did with sci-seq3?  Specifically, we are lucky here in that there are 4 distinct lengths we could have so we could just pad with the 4 distinct nucleotides.  That is, make the barcode length 12 and then:

* If length is 11, add `A`
* If length is 10, add `AC`
* If length is 9, add `AAG`
* If length is 8, add `AAAA`

Or some such.  I'm not sure how much of a difference it would make, but it would be nice potentially to be able to say we don't introduce any collisions this way.

Reply:
Thanks! I have updated this accordingly.